presto-orc/src/main/java/com/facebook/presto/orc/reader/ListStreamReader.java

```diff
@@ -97,35 +97,35 @@ public Block readBlock(Type type)
             }
         }
 
-        // The length vector could be reused, but this simplifies the code below by
-        // taking advantage of null entries being initialized to zero.  The vector
-        // could be reinitialized for each loop, but that is likely just as expensive
-        // as allocating a new array
-        int[] lengthVector = new int[nextBatchSize];
+        // We will use offsetVector vector as a temporary buffer to read from lengthStream
+        int[] offsetVector = new int[nextBatchSize + 1];
         boolean[] nullVector = new boolean[nextBatchSize];
         if (presentStream == null) {
             if (lengthStream == null) {
                 throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is not present");
             }
-            lengthStream.nextIntVector(nextBatchSize, lengthVector);
+            lengthStream.nextIntVector(nextBatchSize, offsetVector);
         }
         else {
             int nullValues = presentStream.getUnsetBits(nextBatchSize, nullVector);
             if (nullValues != nextBatchSize) {
                 if (lengthStream == null) {
                     throw new OrcCorruptionException(streamDescriptor.getOrcDataSourceId(), "Value is not null but data stream is not present");
                 }
-                lengthStream.nextIntVector(nextBatchSize, lengthVector, nullVector);
+                lengthStream.nextIntVector(nextBatchSize, offsetVector, nullVector);
             }
         }
-        int[] offsets = new int[nextBatchSize + 1];
-        for (int i = 1; i < offsets.length; i++) {
-            int length = lengthVector[i - 1];
-            offsets[i] = offsets[i - 1] + length;
+
+        int lastLength = offsetVector[0];
+        offsetVector[0] = 0;
+        for (int i = 1; i < offsetVector.length; i++) {
+            int length = offsetVector[i];
```

Comment:
Ok, I renamed lastLength to currentLength, and length to nextLength. And I think your array index change suggestion makes sense, but it does't affect correctness and I don't like to add another line after the loop. I think with the name nextLength, it would be easy enough for the readers to understand it.